.github/workflows/test-and-deploy.yml

```diff
@@ -171,3 +178,142 @@ jobs:
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      # pin version for poetry as higher versions have PEP 440 versioning issues
+      - name: Install poetry
+        run: pipx install poetry==1.1.15
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ needs.python_version.outputs.version }}
+          cache: "poetry"
+
+      # Poetry must explicitly be told to use the expected version https://github.com/actions/setup-python/issues/374
+      - name: Install dependencies
+        run: |
+          poetry env use ${{ needs.python_version.outputs.version }}
+          poetry config http-basic.properly "${{ secrets.PIP_GEMFURY_DEPLOY_TOKEN }}" NOPASS
+          poetry install
+          npm install
+
+      - name: Compute raw code metrics
+        run: poetry run radon raw ${{ inputs.src_directory }} -j -O radon_raw.json
+
+      - name: Sum raw code metrics
+        id: compute_raw_sum
+        run: |
+          RAW_SUMMARY=$(jq -c "map(to_entries) | add | group_by(.key) | map({ key: .[0].key, value: map(.value) | add }) | from_entries" radon_raw.json)
+          echo "RAW_SUMMARY=${RAW_SUMMARY}" >> $GITHUB_OUTPUT
+          echo ::debug::${RAW_SUMMARY}
+
+      - name: Compute cyclomatic complexity metrics
+        run: poetry run radon cc ${{ inputs.src_directory }} -j -O radon_cc.json
+
+      - name: Count number of files
+        id: compute_cc_file_count
+        run: |
+          CC_FILE_COUNT=$(jq length radon_cc.json)
+          echo "CC_FILE_COUNT=${CC_FILE_COUNT}" >> $GITHUB_OUTPUT
+          echo ::debug::${CC_FILE_COUNT}
+
+      - name: Count number of functions
+        id: compute_cc_function_count
+        run: |
+          CC_FUNCTION_COUNT=$(cat radon_cc.json | grep -o \"function\" | wc -l)
+          echo "CC_FUNCTION_COUNT=${CC_FUNCTION_COUNT}" >> $GITHUB_OUTPUT
+          echo ::debug::${CC_FUNCTION_COUNT}
+
+      - name: Count number of classes
+        id: compute_cc_class_count
+        run: |
+          CC_CLASS_COUNT=$(cat radon_cc.json | grep -o \"class\" | wc -l)
+          echo "CC_CLASS_COUNT=${CC_CLASS_COUNT}" >> $GITHUB_OUTPUT
+          echo ::debug::${CC_CLASS_COUNT}
+
+      - name: Count number of methods
+        id: compute_cc_method_count
+        run: |
+          CC_METHOD_COUNT=$(cat radon_cc.json | grep -o \"method\" | wc -l)
+          echo "CC_METHOD_COUNT=${CC_METHOD_COUNT}" >> $GITHUB_OUTPUT
+          echo ::debug::${CC_METHOD_COUNT}
+
+      - name: Report metrics to Datadog
+        uses: masci/datadog@v1
+        with:
+          api-key: ${{ secrets.DD_API_KEY }}
+          metrics: |
+            - type: "count"
+              name: "properly.ci.code.metrics.loc"
+              value: ${{ fromJSON( steps.compute_raw_sum.outputs.RAW_SUMMARY ).loc }}
+              host: ${{ github.repository_owner }}
+              tags:
+                - "repository:${{ github.repository }}"
+                - "workflow:${{ github.workflow }}"
```

Comment:
Unsure if this is a useful tag

Reply:
Is there any danger in having additional tags? Can we remove it later on if we don't find it useful?

Reply:
Nope! Other than cost spend. Unique combinations of metric names and tags count as a billable metric.

Seems like a reversible decision, now with the reusable workflows 😄 I will leave it in for now